### PR TITLE
Add examples for using spring-data-opensearch w/o Spring Boot

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Add examples for using spring-data-opensearch w/o Spring Boot

### Issues Resolved
Closes https://github.com/opensearch-project/spring-data-opensearch/issues/58

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spring-data-opensearch-examples/spring-boot-gradle/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Spring Data OpenSearch Spring Boot Example Project
+=== 
+
+This sample project demonstrates the usage of the [Spring Data OpenSearch](https://github.com/opensearch-project/spring-data-opensearch/) in the typical Spring Boot web application. The application assumes that there is an [OpenSearch](https://opensearch.org) service up and running on the local machine, available at `https://localhost:9200` (protected by basic authentication with default credentials).
+
+1. The easiest way to get [OpenSearch](https://opensearch.org) service up and running is by using [Docker](https://www.docker.com/):
+
+```shell
+docker run -p 9200:9200 -e "discovery.type=single-node" opensearchproject/opensearch:2.4.0
```
Comment (reta) [inline spring-data-opensearch-examples/spring-boot-gradle/README.md:9]: Aligning with https://github.com/opensearch-project/spring-data-opensearch/pull/63

File: spring-data-opensearch-examples/spring-boot-gradle/src/test/java/org/opensearch/data/example/repository/MarketplaceRepositoryIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+@Testcontainers(disabledWithoutDocker = true)
+@ImportAutoConfiguration(
+        exclude = ElasticsearchDataAutoConfiguration.class,
+        classes = {
+            OpenSearchRestClientAutoConfiguration.class,
+            OpenSearchRestHighLevelClientAutoConfiguration.class,
+            OpenSearchDataAutoConfiguration.class
+        })
+@DataElasticsearchTest
```
Comment (reta) [inline spring-data-opensearch-examples/spring-boot-gradle/src/test/java/org/opensearch/data/example/repository/MarketplaceRepositoryIntegrationTests.java:37]: @dlvenable @dblock question here guys, we could not make `@DataElasticsearchTest` to work seamlessly with OpenSearch (the changes have to be done in Spring Boot and had been rejected previously, no point to try it again), so as an option, we could add support of the `@DataOpenSearchTest` (and consequently, test module to support that), thoughts?

Comment (dblock) [reply spring-data-opensearch-examples/spring-boot-gradle/src/test/java/org/opensearch/data/example/repository/MarketplaceRepositoryIntegrationTests.java:37]: 100%, we should add that here

**Review by dblock — APPROVED** [review]

**Review by dblock — APPROVED** [review]

File: spring-data-opensearch-examples/spring-boot-gradle/src/test/java/org/opensearch/data/example/repository/MarketplaceRepositoryIntegrationTests.java (hunk 2 above)
Comment (reta) [reply spring-data-opensearch-examples/spring-boot-gradle/src/test/java/org/opensearch/data/example/repository/MarketplaceRepositoryIntegrationTests.java:37]: https://github.com/opensearch-project/spring-data-opensearch/issues/71

**Pull request merged**